script.js

```diff
@@ -1,147 +1,199 @@
-        resultsDiv.innerHTML = '';
-
-            // Reverse the results if isBackwards is true to ensure they are displayed from left to right
-            if (isBackwards) {
-                results.reverse();
-            }
-
-            results.forEach((result, i) => {
-                const groupDiv = document.createElement('div');
-                groupDiv.classList.add('result-group');
-
-                const matchedPartSpan = document.createElement('span');
-                matchedPartSpan.classList.add('matched-part');
-                matchedPartSpan.textContent = result.matchedPart.toUpperCase();
-                groupDiv.appendChild(matchedPartSpan);
-
-                result.stickers.forEach(sticker => {
-                    const stickerWrapper = document.createElement('div');
-                    stickerWrapper.classList.add('sticker-wrapper');
-                    stickerWrapper.setAttribute('data-name', sticker.name); // Set the tooltip text in data attribute
-
-                    const image = document.createElement('img');
-                    image.src = sticker.image;
-                    image.alt = sticker.name;
-                    image.classList.add('sticker-image');
-
-                    stickerWrapper.onclick = () => {
-                        // Add sticker info to selectedStickers
-                        selectedStickers[i].sticker = sticker
-                        selectedStickers[i].index = i
-                        renderSelectedStickers(selectedStickers)
-                    };
-
-                    stickerWrapper.appendChild(image);
-                    groupDiv.appendChild(stickerWrapper);
-                });
-
-                selectedStickers.push({matchedPart: result.matchedPart}) // Create object for each matched part
-                resultsDiv.appendChild(groupDiv);
-            });
-
-            const selectedStickersList = document.getElementById('selectedStickers');
-
-            if (results.length <= 0) {
-                const message = results.map(x => x.matchedPart).join('') !== inputVal
-                    ? "Could not match the entire input."
-                    : "No matches found for your input."
-
-                displayInfoMessage(resultsDiv, message, inputVal)
-
-                selectedStickersList.style.display = "none"
-            }  else {
-                selectedStickersList.style.display = "block"
-            }
-            renderSelectedStickers(selectedStickers)
-    } catch (error) {
-        console.error('Error fetching data:', error);
-        alert('Failed to fetch data from the worker.');
-    }
+  let inputVal = document.getElementById("stickerInput").value
+  // Remove spaces from the input
+  inputVal = inputVal.replace(/\s+/g, "")
+  const isBackwards = document.getElementById("isBackwards").checked
+  const isDepth = document.getElementById("isDepth").checked
+
+  const apiUrl = `http://localhost:8787/?input=${encodeURIComponent(
```

Comment:
This should be changed the the actual worker URL.
```suggestion
  const apiUrl = `https://polished-bush-fd60.cryck.workers.dev/?input=${encodeURIComponent(
```